bittensor/_config/__init__.py

```diff
@@ -37,13 +40,15 @@ class InvalidConfigFile(Exception):
         """ In place of YAMLError
         """
 
-    def __new__( cls, parser: ArgumentParser = None, strict: bool = False ):
+    def __new__( cls, parser: ArgumentParser = None, strict: bool = False, args: Optional[List[str]] = None ):
```

Comment:
Is there a reason for this change? I will be hesitant to change the base config class unless necessary. Every single bittensor object initiates a base config object that is the main transfer of information from one class to another.

Reply:
I also think the config class should be updated soon, so it might be good for the next update.

Reply:
this is a dependency injection of the sys.argv from the command line. 

Normally argparser gets it internally, if you don't specify which args to parse, by calling sys.argv the same way I added in the neuron `__main__`

I added it as an optional arg and lower down it defaults to calling sys.argv if None, so the class will function as normal without the arg specified.

The dependency injection is important for testing if `btcli --help` works correctly and has the text we expect. Otherwise I would have to patch argparser, which is subject to API changes. This is recommended from some SO posts I read.

Reply:
Since this sounds like a temporary change before the cleanup, i would generate some issues in this repository to not forget about the tech debt we have around this code.